### PR TITLE
Fix ability to parse empty arrays into an allocated slice

### PR DESCRIPTION
**Pull request by alexpjohnson**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1 +1,3 @@
-module github.com/mitchellh/mapstructure
```
Comment (jamesatha) [inline go.mod:1]: Are we planning to make a pull request up to that repo? Should we make a separate PR with this change so we can at least attempt a PR into that repo?

Comment (alexpjohnson) [reply go.mod:1]: The PR that implements this change, https://github.com/mitchellh/mapstructure/pull/155, has been open since February.

**Review by jamesatha — APPROVED** [review]
